src/common/utils/storage.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+
+// a single place to define all used localStorage keys, to avoid collisions or hardcoded strings
+export enum StorageKeys {
+  COUNTY_MAP_TYPE = 'COUNTY_MAP_TYPE',
+  FULLSTORY_RECORD = 'FULLSTORY_RECORD',
+}
+
+// From https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#testing_for_availability
+export function storageAvailable(storageType: string) {
+  let storage;
+  const testKey = '__storage_test__';
+  try {
+    // @ts-ignore
+    storage = window[storageType];
```

Comment:
interesting thing here is that it's testing two things, feature availability and space availability.
it seems a little moot to test for space (the `QuotaExceededError`) since you still have to try/catch below (the sizes of the objects we save aren't necessarily the same as the `__storage_test__` key above. 

so, to that end, seems like, this feature availability could be memoized - wdyt? try/catching to see whether there's still space available in the localStorage has to be done with every set anyway ¯\_(ツ)_/¯